src/Raven.Server/Commercial/LicenseHelper.cs

```diff
@@ -0,0 +1,230 @@
+            {
+                return DeserializeLicense(licenseString);
+            }
+            catch (Exception e)
+            {
+                var msg = $"Failed to read license from '{LicenseStringConfigurationName}' configuration.";
+
+                if (Logger.IsInfoEnabled)
+                    Logger.Info(msg, e);
+
+                if (throwOnFailure)
+                    throw new LicenseActivationException(msg, e);
+            }
+
+            return null;
+        }
+
+        public License TryGetLicenseFromPath(bool throwOnFailure)
+        {
+            var path = _serverStore.Configuration.Licensing.LicensePath;
+
+            try
+            {
+                using (var stream = File.Open(path.FullPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    return DeserializeLicense(stream);
+                }
+            }
+            catch (Exception e)
+            {
+                var msg = $"Failed to read license from '{path.FullPath}' path.";
+
+                if (Logger.IsInfoEnabled)
+                    Logger.Info(msg, e);
+
+                if (throwOnFailure)
+                    throw new LicenseActivationException(msg, e);
+            }
+
+            return null;
+        }
+
+        private static License DeserializeLicense(Stream stream)
+        {
+            using (var context = JsonOperationContext.ShortTermSingleUse())
+            {
+                var json = context.Read(stream, "license/json");
+                return JsonDeserializationServer.License(json);
+            }
+        }
+
+        private static License DeserializeLicense(string licenseString)
+        {
+            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(licenseString)))
+            {
+                return DeserializeLicense(stream);
+            }
+        }
+
+        private static void UpdateEnvironmentVariableLicenseString(License newLicense, RSAParameters rsaParameters)
+        {
+            var preferredTarget = EnvironmentVariableTarget.Machine;
+            var licenseString = GetLicenseString();
+            if (licenseString == null)
+            {
+                preferredTarget = EnvironmentVariableTarget.User;
+                licenseString = GetLicenseString();
+            }
+
+            if (licenseString == null)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failed to access the environment variable: `{RavenConfiguration.EnvironmentVariableLicenseString}`");
+                return;
+            }
+
+            var oldLicense = DeserializeLicense(licenseString);
+            if (ValidateLicense(newLicense, rsaParameters, oldLicense) == false)
+                return;
+
+            var newLicenseString = JsonConvert.SerializeObject(newLicense);
+            Environment.SetEnvironmentVariable(RavenConfiguration.EnvironmentVariableLicenseString, newLicenseString, preferredTarget);
```

Comment:
This is non persistent set.
For that matter, I'm not sure that there is something that we _can_ do properly here.
Consider a docker instance, which may have the env setup externally completely.

Shouldn't we keep a local copy of the license inside the server store instead as well?